presto-main/src/main/java/io/prestosql/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -620,7 +620,14 @@ public PlanWithProperties visitUnion(UnionNode node, HashComputationSet parentPr
         public PlanWithProperties visitProject(ProjectNode node, HashComputationSet parentPreference)
         {
             Map<Symbol, Symbol> outputToInputMapping = computeIdentityTranslations(node.getAssignments().getMap());
-            HashComputationSet sourceContext = parentPreference.translate(symbol -> Optional.ofNullable(outputToInputMapping.get(symbol)));
+            ImmutableMap.Builder<HashComputation, HashComputation> builder = ImmutableMap.builder();
+            for (HashComputation hashComputation : parentPreference.getHashes()) {
```

Comment:
This seems *almost* what `HashComputationSet::translate` does, except that it creates a map instead of a set. Maybe we can generalize what that method does to serve this purpose?